CONTRIBUTING.md

```diff
@@ -0,0 +1,108 @@
+# Mintlayer core contributing guide
+
+We are happy to take contributions to the project in any form: if you find a bug feel free to create an issue, if you fix a bug feel free to create a pr to merge your fix, if you want to add a totally new feature go ahead and do so.
+
+
+## Setup
+
+Mintlayer is built using Substrate which means you'll need rust and substrate set up to do any development work. The functional tests require python to be installed too.
+
+You can follow these steps below or go Substrate's own guide [here](https://docs.substrate.io/v3/getting-started/overview/)
+### Build essentials
+
+#### Ubuntu
+    sudo apt update
+    sudo apt install -y git clang curl libssl-dev llvm libudev-dev
+
+#### Fedora
+    sudo dnf update
+    sudo dnf install clang curl git openssl-devel
+
+#### Arch
+    pacman -Syu --needed --noconfirm curl git clang
+    
+#### MacOS (Note there are issues with M1 processors - you're on your own for now :D)
+    brew update
+    brew install openssl
+    
+
+### Rust
+    curl https://sh.rustup.rs -sSf | sh
+    source ~/.cargo/env
+    rustup default stable
+    rustup update
+    rustup update nightly
+    rustup target add wasm32-unknown-unknown --toolchain nightly
+
+### Functional test stuff
+Ensure python3 is already installed. I'll be shocked if it isn't. After that install the dependencies listed below.
+
+    python -m pip install 'substrate-interface == 0.13.12'
+    python -m pip install 'scalecodec == 0.11.18'
+    
+## Building Mintlayer
+
+To build Mintlayer go to [the main repo](https://github.com/mintlayer/core) and clone the repo using git like you would with any other repo.
+Development happens on the staging branch so make sure you can build Mintlayer by checking out the staging branch and building the codebase.
+
+    git fetch --all
+    git checkout staging
+    cargo build
+    
+The above will build the code base in debug mode. If you go to `target/debug` you should find the `mintlayer-core` binary. If you have issues building it's probably 
+the installation steps above, double-check them and try again. If the build keeps failing then feel free to reach out to us in an issue or on [discord](https://discord.gg/XMrhvngQ8T).
+
+If that all goes swimmingly then it's time to do something very very slightly more exciting. Make sure you can run the unit tests using `cargo test`
+   
+Yep wasn't that exciting. All of those tests should pass, if they don't feel free to dig in and find out what's going on.
+
+To build a release version of Mintlayer you simply run `cargo build --release`
+    
+To run the functional tests, ensure Mintlayer has been compiled first, go to `test` and run `test_runner.py`. Everything else will happen automatically.
+
+## How to actually contribute
```

Comment:
Should we have a mention about responsible disclosure here? I don't think we want people reporting security issues directly on Github

Reply:
Yeah I was planning to update this once the bug bounty page is up on the website along with the gpg keys. I'll add a paragraph now (probably to the README rather than here so it's more obvious). Maybe I should add a file like SECURITY.md too?